File: frontend/src/app/shared/service/collaborator/recomrecla/ReclamationEtudiantCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {ReclamationEtudiantDto} from '../../../model/recomrecla/ReclamationEtudiant.model';
import {ReclamationEtudiantCriteria} from '../../../criteria/recomrecla/ReclamationEtudiantCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class ReclamationEtudiantCollaboratorService extends AbstractService<ReclamationEtudiantDto, ReclamationEtudiantCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/reclamationEtudiant/';
    }

    public constrcutDto(): ReclamationEtudiantDto {
        return new ReclamationEtudiantDto();
    }

    public constrcutCriteria(): ReclamationEtudiantCriteria {
        return new ReclamationEtudiantCriteria();
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/recomrecla/ReclamationEtudiantCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.recomrecla;

import java.util.List;
import ma.zs.alc.bean.core.recomrecla.ReclamationEtudiant;
import ma.zs.alc.dao.criteria.core.recomrecla.ReclamationEtudiantCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface ReclamationEtudiantCollaboratorService extends  IService<ReclamationEtudiant,ReclamationEtudiantCriteria>  {

    List<ReclamationEtudiant> findByTypeReclamationEtudiantId(Long id);
    int deleteByTypeReclamationEtudiantId(Long id);
    long countByTypeReclamationEtudiantCode(String code);




}
